### PR TITLE
Changes the client Tags All() method to follow links

### PR DESCRIPTION
**Pull request by BrianBland**
This returns all tags even when the registry forces pagination.

### CODE REVIEW

**Comment by RichardScothern** [comment]
😸  hello Brian.

Any chance of a unit test?

**Comment by BrianBland** [comment]
Required some janky hacks, but sure thing 🍰

**Comment by RichardScothern** [comment]
🎩

File: registry/client/repository.go [hunk 1]
```diff
@@ -213,28 +214,35 @@ func (t *tags) All(ctx context.Context) ([]string, error) {
...
+
+			tagsResponse := struct {
+				Tags []string `json:"tags"`
+			}{}
+			if err := json.Unmarshal(b, &tagsResponse); err != nil {
+				return tags, err
+			}
+			tags = append(tags, tagsResponse.Tags...)
+			if link := resp.Header.Get("Link"); link != "" {
+				u = strings.Split(link, ";")[0]
```
Comment (RichardScothern) [inline registry/client/repository.go]: The link should be surrounded by `<>` as per RFC 5988.  I think you should add to your unit tests and strip them out here.

File: registry/client/repository_test.go [hunk 2]
```diff
@@ -949,6 +950,78 @@ func TestManifestTags(t *testing.T) {
...
+		if i > 0 {
+			queryParams["n"] = []string{"1"}
+			queryParams["last"] = []string{tagsList[i-1]}
+		}
+		headers := http.Header(map[string][]string{
+			"Content-Length": {fmt.Sprint(len(body))},
+			"Last-Modified":  {time.Now().Add(-1 * time.Second).Format(time.ANSIC)},
+		})
+		if i < 2 {
+			headers.Set("Link", s.URL+"/v2/"+repo.Name()+"/tags/list?n=1&last="+tagsList[i]+`; rel="next"`)
```
Comment (RichardScothern) [inline registry/client/repository_test.go]: Same comment as above..

File: registry/client/repository.go (hunk 1 above)
Comment (RichardScothern) [inline registry/client/repository.go]: Note that curl doesn't object to them being there for some reason.

Comment (BrianBland) [inline registry/client/repository.go]: Interesting... I'll strip them out, but I know of at least one registry implementation which does not use the chevrons 😞

**Comment by RichardScothern** [comment]
LGTM

**Comment by dmcgowan** [comment]
LGTM

**Comment by linlinlinlin** [comment]
Is this PR can get image's "all" tags just sent one request without auth?

**Pull request merged**